megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java

```diff
@@ -0,0 +1,1242 @@
+            filterToks.add(new OperationFT(MechSearchFilter.BoolOp.AND));
+            txtEqExp.setText(filterExpressionString());
+            btnBack.setEnabled(true);
+            disableOperationButtons();
+            enableSelectionButtons();
+        } else if (ev.getSource().equals(btnOr)) {
+            filterToks.add(new OperationFT(MechSearchFilter.BoolOp.OR));
+            txtEqExp.setText(filterExpressionString());
+            btnBack.setEnabled(true);
+            disableOperationButtons();
+            enableSelectionButtons();
+        } else if (ev.getSource().equals(btnBack)) {
+            if (!filterToks.isEmpty()) {
+                filterToks.remove(filterToks.size() - 1);
+                txtEqExp.setText(filterExpressionString());
+                if (filterToks.isEmpty()) {
+                    btnBack.setEnabled(false);
+                }
+
+                if ((filterToks.isEmpty()) || (filterToks.lastElement() instanceof OperationFT)) {
+                    disableOperationButtons();
+                    enableSelectionButtons();
+                } else {
+                    enableOperationButtons();
+                    disableSelectionButtons();
+                }
+            }
+        } else if (ev.getSource().equals(btnClear)) {
+            filterToks.clear();
+            txtEqExp.setText("");
+            btnBack.setEnabled(false);
+            disableOperationButtons();
+            enableSelectionButtons();
+        }
+    }
+
+    private boolean matchTechLvl(int t1, int t2) {
+        return ((t1 == TechConstants.T_ALL) || (t1 == t2)
+                || ((t1 == TechConstants.T_IS_TW_ALL) && (t2 <= TechConstants.T_IS_TW_NON_BOX)))
+
+                || ((t1 == TechConstants.T_TW_ALL) && (t2 <= TechConstants.T_CLAN_TW))
+
+                || ((t1 == TechConstants.T_ALL_IS) && ((t2 <= TechConstants.T_IS_TW_NON_BOX)
+                || (t2 == TechConstants.T_IS_ADVANCED)
+                || (t2 == TechConstants.T_IS_EXPERIMENTAL)
+                || (t2 == TechConstants.T_IS_UNOFFICIAL)))
+
+                || ((t1 == TechConstants.T_ALL_CLAN)
+                && ((t2 == TechConstants.T_CLAN_TW)
+                || (t2 == TechConstants.T_CLAN_ADVANCED)
+                || (t2 == TechConstants.T_CLAN_EXPERIMENTAL)
+                || (t2 == TechConstants.T_CLAN_UNOFFICIAL)));
+    }
+
+    private boolean matchTechClass(String t1, String t2) {
+        if (t1.equals("All")) {
+            return true;
+        } else if (t1.equals("IS/Clan")) {
+            return t2.equals("Inner Sphere") || t2.equals("Clan") || t1.equals(t2);
+        } else {
+            return t1.equals(t2);
+        }
+    }
+
+    private boolean matchUnitType(int unitTypeFilter, EquipmentType eq) {
+        // All is selected
+        if (unitTypeFilter < 0) {
+            return true;
+        }
+
+        switch (unitTypeFilter) {
+            case 5: //UnitType.AERO: the aero index is out of order
```

Comment:
```suggestion
            case 5: // UnitType.AERO: the aero index is out of order
```
Any idea why we aren't using the constant?

Reply:
No idea. I really didnt look at this code at all beyond making it work as a panel instead of the full dialog. But UnitType.AERO is 9 and this checks for 5 so it's probably better that way :)